src/gateway/WebsocketShard.ts

```diff
@@ -45,7 +45,7 @@ class WebsocketShard {
   private globalBucket: Bucket = new Bucket(120, 60000);
   private presenceUpdateBucket: Bucket = new Bucket(5, 60000);
 
-  public constructor(private token: string, private client: Client) {}
```

Comment:
Maybe we should pass what we need, instead of the whole client. This would allow our modules to be less cross dependent on each other.

Reply:
that would be a subject for the later API, I'd not worry about this when this gets into master